### ISSUE TITLE
[FP] A === B seems to be always false

### ISSUE THREAD

**Issue by artjomsimon** [kalessil/phpinspectionsea#1578]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) 3.0.9.1 |
| Language level | PHP 7.*                        |

```php
<?php declare(strict_types=1);

class A {

    public $prop;

    public function isEqualTo(A $a) : bool {
        return $this->prop === $a->prop;
    }

    public function isDifferentFrom(A $a) : bool {
        return $this->isEqualTo($a) === false;  // [EA] '$this->isEqualTo($a) === false' seems to be always false. 
    }
}
```

The warning about A::isDifferentFrom() always being false is a false positive.

**Comment by kalessil** [comment]
@artjomsimon: can not reproduce, can you try Invalidating caches and checking the issue again?

**Comment by artjomsimon** [comment]
@kalessil can reproduce with latest IntelliJ IDEA 2020.2 EAP (Ultimate Edition) and EA Ultimate Plugin 3.0.9.1 after a `File->Invalidate Caches and Restart`

```
Build #IU-202.6109.22, JAVA Runtime version: 11.0.7+10-b944.16 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.7.7-arch1-1
```

**Comment by kalessil** [comment]
Okay, that's 2020.2 specific, I made a forward compatibility fix - will release the change somewhere this week.
